File: ColdDream.Api/Models/CustomTour.cs
using System.ComponentModel.DataAnnotations;

namespace ColdDream.Api.Models;

public class CustomTour
{
    public Guid Id { get; set; }
    
    public Guid UserId { get; set; }
    
    [Required]
    public string Destination { get; set; } = string.Empty;
    
    [Required]
    public string StartDate { get; set; } = string.Empty;
    
    [Required]
    public int Days { get; set; }
    
    [Required]
    public int PeopleCount { get; set; }
    
    [Required]
    public string Budget { get; set; } = string.Empty;
    
    public string Requirements { get; set; } = string.Empty;
    
    public string ContactName { get; set; } = string.Empty;
    
    public string ContactPhone { get; set; } = string.Empty;
    
    public string Status { get; set; } = "Pending"; // Pending, Processing, Completed
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: ColdDream.Web/src/api/custom-tour.ts
import api, { type ApiResponse } from './index';

export interface CustomTour {
  id: string;
  destination: string;
  startDate: string;
  days: number;
  peopleCount: number;
  budget: string;
  requirements: string;
  contactName: string;
  contactPhone: string;
  status: string;
  createdAt: string;
}

export interface CreateCustomTourDto {
  destination: string;
  startDate: string;
  days: number;
  peopleCount: number;
  budget: string;
  requirements: string;
  contactName: string;
  contactPhone: string;
}

export const createCustomTour = (data: CreateCustomTourDto) => {
  return api.post<any, ApiResponse<CustomTour>>('/customtour', data);
};

export const getMyCustomTours = () => {
  return api.get<any, ApiResponse<CustomTour[]>>('/customtour/my');
};

export const getCustomTourById = (id: string) => {
  return api.get<any, ApiResponse<CustomTour>>(`/customtour/${id}`);
};

export const cancelCustomTour = (id: string) => {
  return api.post<any, ApiResponse<any>>(`/customtour/${id}/cancel`); // Assuming backend supports this
};
